### PR TITLE
Check for maker on config level for start mode

### PR DESCRIPTION
**Pull request by katleiahramos**
Previously, maker toolkit blocks weren't showing when editing in start mode even though the maker toolkit API was enabled. This change ensures that maker toolkit shows in start mode. 

## Links
- [jira ticket](https://codedotorg.atlassian.net/browse/STAR-1688)

## Testing story
Open to ideas of how to add a UI test for this change. 

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/StudioApp.js [hunk 1]
```diff
@@ -2279,7 +2279,8 @@ StudioApp.prototype.handleEditCode_ = function(config) {
   }
 
   // Remove maker API blocks from palette, unless maker APIs are enabled.
-  if (!project.getMakerAPIs()) {
+  // Also check config.level to ensure maker toolkit shows when editing start mode
```
Comment (maddiedierker) [inline apps/src/StudioApp.js]: i think this comment is misleading -- `config.level.makerlabEnabled` and start mode should be decoupled, so maybe this should be something like "any level with maker enabled"

File: apps/src/StudioApp.js [hunk 2]
```diff
@@ -2279,7 +2279,8 @@ StudioApp.prototype.handleEditCode_ = function(config) {
...
+  if (!project.getMakerAPIs() && !config.level.makerlabEnabled) {
```
Comment (maddiedierker) [inline apps/src/StudioApp.js]: what happens in the following situation? is it even possible?
1. a level has maker enabled
2. the user disables maker from the "settings" cog
3. reload the page

my assumption would be that `project.getMakerAPIs()` returns falsy, but `config.level.makerlabEnabled` is still true, which means the maker blocks would still display in the toolbox, which i think is unexpected

Comment (katleiahramos) [reply apps/src/StudioApp.js]: hmm great point! I'll look into it

**Review by maddiedierker — APPROVED** [review]

File: apps/src/StudioApp.js (hunk 2 above)
Comment (katleiahramos) [reply apps/src/StudioApp.js]: Added logic to make check if the level is specifically `inStartMode`! Also updated the comment above, let me know if it makes sense:

https://github.com/code-dot-org/code-dot-org/pull/44156/commits/aa2de090f614a7e7d983a7d8d7373f162f641fc5

**Comment by katleiahramos** [comment]
For some reason this unit test keeps failing in my CI run. It seems unrelated to my changes, so I tried re-running it a few times but it is definitely consistently failing. Any ideas?

```
test_render_pegasus_documents                                   FAIL (138.45s)
--
13018 | Minitest::Assertion:         Page rendering errors:
13019 | [code.org/about/team] Render failed:
13020 | Error rendering /drone/src/pegasus/sites.v3/code.org/views/about_headshots.haml: undefined method `downcase' for nil:NilClass
13021 | /drone/src/pegasus/sites.v3/code.org/public/about/team.md.erb
13022 | /drone/src/pegasus/router.rb:538:in `rescue in render_template'
13023 | /drone/src/pegasus/router.rb:534:in `render_template'
13024 | /drone/src/pegasus/router.rb:627:in `view'
13025 | /drone/src/lib/cdo/pegasus/actionview_sinatra.rb:50:in `method_missing'
13026 | inline template:6:in
```

**Pull request merged**